runtime/sys.c

```diff
@@ -632,7 +632,7 @@ int caml_unix_random_seed(intnat data[16])
   int nread = 0;
 
   /* Try kernel entropy first */
-#ifdef HAS_GETENTROPY
+#if defined(HAS_GETENTROPY) && !defined(__ANDROID__)
```

Comment:
Rather than having this here, the logic should be in `configure.ac`, making sure that `HAS_GETENTROPY` is not defined if not supported.